text/src/main/java/net/skyscanner/backpack/text/BpkText.kt

```diff
@@ -0,0 +1,60 @@
+package net.skyscanner.backpack.text
+
+import android.content.Context
+import android.content.res.TypedArray
+import android.support.v4.widget.TextViewCompat
+import android.support.v7.widget.AppCompatTextView
+import android.util.AttributeSet
+
+
+open class BpkText(
+  context: Context,
+  attrs: AttributeSet?,
+  defStyleAttr: Int) : AppCompatTextView(context, attrs, defStyleAttr) {
+
+  companion object {
+    val XS = 0
+    val SM = 1
+    val BASE = 2
+    val LG = 3
+    val XL = 4
+    val XXL = 5
+  }
+
+  var textStyle: Int = BASE;
```

Comment:
Shouldn't this be an enum type instead of just `Int`? It seems with this solution it's valid to do something like `text.textStyle = 287319` which is nonsensical. With an enum the compiler would prevent that right?

Reply:
There is a performance oveahead in android for enums and it is recommended to use constants for such cases. This pattern can be found everywhere in android codebase. The concern of having wrong value set to the type is addressed using the annotation `@IntDef`. For example see the linear layout implementation here https://github.com/aosp-mirror/platform_frameworks_base/blob/faac4ad153f55b1d224e2e800c24d0b9eb0ae0e5/core/java/android/widget/LinearLayout.java#L91-L96

Reply:
I've implemented that under Amol's guidance so I'm sure is the best approach 😄